jupiter-notebook/README.md

```diff
@@ -0,0 +1,93 @@
+# Qiskit Development Environment with Docker
+
+This project sets up a Qiskit development environment using [Jupyter Notebook](https://jupyter.org) on Docker. The container is based on jupyter/minimal-notebook image, and, has Qiskit and some useful packages (check `requirements.txt`), allowing you to develop quantum programs locally and execute them inside the container.
+
+## Prerequisites
+
+Make sure you have the following tools installed on your system:
+
+-   [Docker](https://docs.docker.com/get-docker/)
+-   [Docker Compose](https://docs.docker.com/compose/install/)
+
+## Setup instructions
+
+> For a fast installation you can run the script `bash setup.sh`
+
+### 0. Clone this repository
+
+Firstly, you need to clone this repository:
+
+```bash
+git clone git@github.com:unipr-org/QC.git
+```
+
+### 1. Create required folders
+
+```bash
+cd jupiter-notebook
+mkdir workdir
+```
+
+> Note that the `workdir` folder will be mounted inside the container, and it will be used to store the notebooks.
+
+
+### 2. Setup group ownership
+
+Usually the user and group of the host and the container are different. To avoid permission issues, you can change the group ownership of the shared folders to the docker group.
+
+```bash
+chown -R :docker $example_dir $workdir_dir
+chmod -R 774 $example_dir $workdir_dir
+```
+
+### 3. Build the Docker container
+
+To build the Docker image and set up the container without running it immediately, use the following command:
+
+```bash
+docker-compose up --build --no-start --no-cache
+```
+
+> This command will create the container but will **not** start it automatically.
+
+### 4. Start the container
+
+Once the build is complete, you can start the container using:
+
+```bash
+docker-compose up -d
+```
+
+> This command will start the container in the background, and share sessions with the host on port 8888.
+
+### 5. Stop and remove the container
+
+To stop container, use the following command:
+
+```bash
+docker-compose down
```

Comment:
È sbagliato, `docker-compose stop` mette in stop il container, `docker-compose down` lo elimina. Questa parte va lasciata com'era prima.